### ISSUE TITLE
NPE on MMS group message receipt

### ISSUE THREAD

**Issue by junitas** [signalapp/Signal-Android#7493]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
I encountered this shortly after seeing #7492 and the two seem related. 

Shortly after receiving a message from an MMS group Signal got caught in a crash loop. Signal would crash, come back up to the conversation list, and then crash again shortly after. Once I deleted the MMS group conversation and the additional "duplicate" conversations the crashes stopped. The crashes were caused by a NullPointerException related to the MMS messages as can be seen in the debug log. 

```
03-09 20:27:03.322 16906 16924 E AndroidRuntime: FATAL EXCEPTION: JobConsumer-2 
03-09 20:27:03.322 16906 16924 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 16906
03-09 20:27:03.322 16906 16924 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.mms.pdu_alt.GenericPdu.getMmsVersion()' on a null object reference 
03-09 20:27:03.322 16906 16924 E AndroidRuntime: at org.thoughtcrime.securesms.mms.IncomingLollipopMmsConnection.retrieve(IncomingLollipopMmsConnection.java:98) 
03-09 20:27:03.322 16906 16924 E AndroidRuntime: at org.thoughtcrime.securesms.mms.CompatMmsConnection.retrieve(CompatMmsConnection.java:77) 
03-09 20:27:03.322 16906 16924 E AndroidRuntime: at org.thoughtcrime.securesms.jobs.MmsDownloadJob.onRun(MmsDownloadJob.java:118) 
03-09 20:27:03.322 16906 16924 E AndroidRuntime: at org.thoughtcrime.securesms.jobs.MasterSecretJob.onRun(MasterSecretJob.java:18) 
```

### Steps to reproduce
- Receive an MMS group message and possibly observe Signal crashing

**Actual result:** Signal gets caught in a crash loop on incoming MMS group message
**Expected result:** Signal receives happy MMS group message and does not crash.

### Device info

**Device:** Nexus 6
**Android version:** 7.1.1
**Signal version:** 4.17.0

### Link to debug log
https://debuglogs.org/a792bb6f4d1156f26a9cc10376ce39599fcdf41405dac090bf0bb969f4584395

**Comment by PiCob** [comment]
I guess this has been fixed by https://github.com/signalapp/Signal-Android/commit/40d14698b8acae34be18c8138f623f1739f55e14 and will be included in 4.17.1

**Comment by junitas** [comment]
Yup, good catch. This is the same NPE.
